control-plane-operator/hostedclusterconfigoperator/controllers/node/node_test.go

```diff
@@ -0,0 +1,113 @@
+package node
+
+import (
+	"testing"
+
+	supportutil "github.com/openshift/hypershift/support/util"
+
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestNodeToNodePoolName(t *testing.T) {
+	capiv1.AddToScheme(scheme.Scheme)
+
+	machineNamespace := "test"
+	nodePoolName := "ns/name"
+	machineWithNodePoolAnnotation := &capiv1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: machineNamespace,
+			Name:      "hasNodePoolAnnotation",
+			Annotations: map[string]string{
+				nodePoolAnnotation: nodePoolName,
+			},
+		},
+	}
+	machineWithOutNodePoolAnnotation := &capiv1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: machineNamespace,
+			Name:      "DoNotHaveNodePoolAnnotation",
+		},
+	}
+
+	testCases := []struct {
+		name                 string
+		node                 *corev1.Node
+		expectedNodePoolName string
+		error                bool
+	}{
+		{
+			name: "When MachineAnnotation does not exist in Node it should fail",
+			node: &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						capiv1.ClusterNamespaceAnnotation: machineNamespace,
+					},
+				},
+			},
+			expectedNodePoolName: "",
+			error:                true,
+		},
+		{
+			name: "When ClusterNamespaceAnnotation does not exist in Node it should fail",
+			node: &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						capiv1.MachineAnnotation: machineWithNodePoolAnnotation.Name,
+					},
+				},
+			},
+			expectedNodePoolName: "",
+			error:                true,
+		},
+		{
+			name: "When nodePoolAnnotation does not exist in Machine it should fail",
+			node: &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						capiv1.ClusterNamespaceAnnotation: machineNamespace,
+						capiv1.MachineAnnotation:          "doesNotExist",
+					},
+				},
+			},
+			expectedNodePoolName: "",
+			error:                true,
+		},
+		{
+			name:                 "When Machine does not exist it should fail",
```

Comment:
Seems that this describes the case above, this case looks like "When node has no annotations"

Reply:
updated.